### PR TITLE
Deterministic parsing and empty and nullable actual and expected outputs

### PR DESCRIPTION
**Pull request by neoxelox**
> [!NOTE]
> Improves reliability and clarity of evaluation outputs and configuration.
> 
> - Deterministic JSON stringification via `fast-json-stable-stringify` in `extract` for both actual and expected outputs; treat `null` as empty string and error only on `undefined`
> - Add expected-output parsing and optional `fieldAccessor` to configuration UI; auto-clear accessors when format isn’t accessible; enhanced tooltips and empty-output hints in ResultPanel and test preview
> - Tighten validation in LLM and Rule evaluations to check `expectedOutput === undefined`; better edge-case handling for empty strings in `lexicalOverlap` (substring, rouge, levenshtein) and `numericSimilarity`
> - Update tests to cover deterministic ordering, field access (string/number/boolean/null), and empty/nullable cases; minor util fix uses `String(data)` fallback
> - Minor UI tweaks: Tooltip `maxWidth` and messaging around deterministic stringification
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c329d9b946408d383fb066f9a9320bb748cf855d. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/core/src/services/evaluationsV2/rule/lexicalOverlap.ts [hunk 1]
```diff
@@ -156,7 +156,7 @@ async function run(
 
   if (expectedOutput?.error) {
     throw expectedOutput.error
-  } else if (!metadata.expectedOutput) {
+  } else if (metadata.expectedOutput === undefined) {
```
[the remark this answers is not in the document]
Comment (neoxelox) [reply packages/core/src/services/evaluationsV2/rule/lexicalOverlap.ts:159]: check again buddy

File: packages/core/src/services/evaluationsV2/rule/numericSimilarity.ts [hunk 2]
```diff
@@ -136,7 +136,7 @@ async function run(
 
   if (expectedOutput?.error) {
     throw expectedOutput.error
-  } else if (!metadata.expectedOutput) {
+  } else if (metadata.expectedOutput === undefined) {
```
[the remark this answers is not in the document]
Comment (neoxelox) [reply packages/core/src/services/evaluationsV2/rule/numericSimilarity.ts:139]: fixed
